src/components/TeacherComponents/OnboardingForm/OnboardingForm.tsx

```diff
@@ -51,17 +54,27 @@ const OnboardingForm = ({
   const utmParams = useUtmParams();
   const { posthogDistinctId } = useAnalytics();
   const { user } = useUser();
+  const [submitError, setSubmitError] = useState<string | null>(null);
 
   const onFormSubmit = async (data: OnboardingFormProps) => {
     if ("worksInSchool" in data) {
-      router.push(
-        resolveOakHref({
+      router.push({
+        pathname: resolveOakHref({
           page: data.worksInSchool
             ? "onboarding-school-selection"
             : "onboarding-role-selection",
         }),
-      );
+        query: router.query,
```

Comment:
Maintains the `returnTo` param so that we know which page to return the user to after they have onboarded